beluga_tutorial/src/main.cpp

```diff
@@ -0,0 +1,264 @@
+// Copyright 2022-2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cmath>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <string>
+#include <vector>
+
+#include <range/v3/action.hpp>
+#include <range/v3/algorithm/any_of.hpp>
+#include <range/v3/algorithm/min_element.hpp>
+#include <range/v3/numeric/accumulate.hpp>
+#include <range/v3/range/conversion.hpp>
+#include <range/v3/view/generate.hpp>
+#include <range/v3/view/take_exactly.hpp>
+
+#include <yaml-cpp/yaml.h>
+#include <beluga/beluga.hpp>
+
+struct Parameters {
+  /// Size of the 1D map in (m)
+  /**
+   * In the simulation, the 1D map is continuous,
+   * but the limit is defined as an integer to simplify data visualization.
+   */
+  int map_size{100};
```

Comment:
nit: Use `std::size_t` instead of `int` or `unsigned int` to represent counted quantities.